File: MinjiJu/2022-07-08-unique-paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        
        double res = 1;
        
        // nCr
        // select minimum paths able to take (right/down) from total
        for(int i=1; i<min(n,m); i++){
            res = res*(n+m-i-1)/i;
        }
        return (int)res;
    }
};

File: Chloe Gwon/2022-07-08-Unique-Paths.java
class Solution {
    public int uniquePaths(int m, int n) {

        int count[][] = new int[m][n];
        
        for (int i=0; i<m; i++){
            count[i][0] = 1;
        }
        for (int j=0; j<n; j++){
            count[0][j] = 1;
        }
        
        for (int i=1; i<m; i++){
            for (int j=1; j<n; j++){
                count[i][j] = count[i][j-1] + count[i-1][j];
            }
        }
        
        return count[m-1][n-1];
    }
}
